### DISCUSSION TITLE
unexpected data type conversion

### DISCUSSION THREAD

**Discussion by jxchen01** [Project-MONAI/MONAI#4475] [General]
I am looking at this segmentation tutorial: https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py

It looks like the data is in uint8 (see: https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py#L50)

But, after transformation, it becomes torch.float32 (I manually checked at this step: https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py#L131) 

I am curious how is this data conversion happening? Can we choose to not converting?

**Comment by Nic-Ma** [comment]
Hi @jxchen01 ,

I think maybe `ScaleIntensityd` transform converted it to float32?

Thanks.

**Reply by jxchen01** [reply]
@Nic-Ma , that was my guess, too. But, even I comment out all the transforms, except the `LoadImage` transform, the data returned from the dataloader is still `torch.float32` ... Super strange to me ...

**Reply by jxchen01** [reply]
I guess this might be the root cause: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/io/array.py#L102

I am not sure if this is good for `LoadImage`. In my opinion, the default behavior should be keeping the original data type instead of converting to float32
